test/contracts/external/median.spec.ts

```diff
@@ -0,0 +1,210 @@
+require('module-alias/register');
+
+import * as _ from 'lodash';
+import * as ABIDecoder from 'abi-decoder';
+import * as chai from 'chai';
+import * as setProtocolUtils from 'set-protocol-utils';
+import { Address, Bytes } from 'set-protocol-utils';
+import { BigNumber } from 'bignumber.js';
+
+import ChaiSetup from '@utils/chaiSetup';
+import { BigNumberSetup } from '@utils/bigNumberSetup';
+import {
+  FeedFactoryContract,
+  PriceFeedContract,
+  MedianContract,
+} from '@utils/contracts';
+import { expectRevertError } from '@utils/tokenAssertions';
+import { Blockchain } from '@utils/blockchain';
+import { ZERO } from '@utils/constants';
+import { getWeb3 } from '@utils/web3Helper';
+
+import { OracleWrapper } from '@utils/wrappers/oracleWrapper';
+
+BigNumberSetup.configure();
+ChaiSetup.configure();
+const web3 = getWeb3();
+const FeedFactory = artifacts.require('FeedFactory');
+const { SetProtocolTestUtils: SetTestUtils, SetProtocolUtils: SetUtils } = setProtocolUtils;
+const { expect } = chai;
+const blockchain = new Blockchain(web3);
+const setUtils = new SetUtils(web3);
+
+
+contract('Oracle', accounts => {
+  const [
+    ownerAccount,
+    secondOracleAccount,
+    thirdOracleAccount,
+  ] = accounts;
+
+  let feedFactory: FeedFactoryContract;
+
+  const oracleWrapper = new OracleWrapper(ownerAccount);
+
+  before(async () => {
+    ABIDecoder.addABI(FeedFactory.abi);
+  });
+
+  after(async () => {
+    ABIDecoder.removeABI(FeedFactory.abi);
+  });
+
+  beforeEach(async () => {
+    await blockchain.saveSnapshotAsync();
+  });
+
+  afterEach(async () => {
+    await blockchain.revertAsync();
+  });
+
+  describe('FeedFactory', async () => {
+    beforeEach(async () => {
+      feedFactory = await oracleWrapper.deployFeedFactoryAsync();
+    });
+
+    describe('#create', async () => {
+      async function subject(): Promise<PriceFeedContract> {
+        return await oracleWrapper.deployPriceFeedAsync(feedFactory);
+      }
+
+      it('creates a price feed', async () => {
+        const priceFeed = await subject();
+
+        const peekResponse = await priceFeed.peek.callAsync();
+        const initialValue = peekResponse[0];
+
+        expect(initialValue).to.bignumber.equal(ZERO);
+      });
+    });
+  });
+
+  describe('PriceFeed', async () => {
+    let priceFeed: PriceFeedContract;
+
+    beforeEach(async () => {
+      feedFactory = await oracleWrapper.deployFeedFactoryAsync();
+      priceFeed = await oracleWrapper.deployPriceFeedAsync(feedFactory);
+    });
+
+    describe('#poke', async () => {
+      let subjectPrice: BigNumber;
+      let subjectTimestamp: BigNumber;
+      let subjectCaller: Address;
+
+      beforeEach(async () => {
+        subjectPrice = new BigNumber(10000000);
```

Comment:
Suggestion: use a realistic price? I believe they use 18 decimal places. So $100 is 100 * 10^18.

Reply:
+1